pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,410 @@
+			new_target: LinkTarget<T::AccountId>,
+		) -> DispatchResult {
+			ensure_root(origin.clone())?;
+
+			ensure!(!namespace.len().is_zero(), Error::<T>::InvalidInput);
+			ensure!(!alias.len().is_zero(), Error::<T>::InvalidInput);
+
+			let namespace = namespace.into_inner();
+
+			let alias_index = Self::take_name_index(&alias);
+			ensure!(
+				<Aliases<T>>::contains_key(&namespace, &alias_index),
+				Error::<T>::AliasNotExists
+			);
+			<Aliases<T>>::insert(&namespace, &alias_index, new_target);
+
+			Ok(())
+		}
+
+		/// Delete an alias.
+		///
+		/// Only the owner of the namespace linked to the alias can execute this.
+		///
+		/// - `namespace`: namespace related to the alias to delete
+		/// - `alias`: the alias to delete
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+		pub fn delete_alias(
+			origin: OriginFor<T>,
+			namespace: BoundedVec<u8, <T as pallet::Config>::NameLimit>,
+			alias: BoundedVec<u8, <T as pallet::Config>::NameLimit>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			ensure!(!namespace.len().is_zero(), Error::<T>::InvalidInput);
+			ensure!(!alias.len().is_zero(), Error::<T>::InvalidInput);
+
+			let namespace = namespace.into_inner();
+
+			let owner = <Namespaces<T>>::get(&namespace).ok_or(Error::<T>::NamespaceNotFound)?;
+			ensure!(&who == &owner, Error::<T>::NotAllowed);
+
+			let alias_index = Self::take_name_index(&alias);
+			ensure!(
+				<Aliases<T>>::contains_key(&namespace, &alias_index),
+				Error::<T>::AliasNotExists
+			);
+
+			<Aliases<T>>::remove(&namespace, &alias_index);
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Utility function that checks for the existence of a name in storage and return its index.
+		///
+		/// - `name`: the reference of the name to lookup
+		///
+		/// Returns:
+		/// - `Compact<u64>`: the index of the name in storage
+		pub fn take_name_index(name: &Vec<u8>) -> Compact<u64> {
+			let name_index = <Names<T>>::get(name);
+			if let Some(name_index) = name_index {
+				<Compact<u64>>::from(name_index)
+			} else {
+				let next_name_index = <NamesIndex<T>>::try_get().unwrap_or_default() + 1;
+				let next_name_index_compact = <Compact<u64>>::from(next_name_index);
+				<Names<T>>::insert(name, next_name_index_compact);
+				// storing is dangerous inside a closure
+				// but after this call we start storing..
+				// so it's fine here
+				<NamesIndex<T>>::put(next_name_index);
+				next_name_index_compact
+			}
+		}
+
+		/// Utility function that checks if an AccountId is the owner of the assets in the LinkTarget
+		///
+		/// - `who`: the AccountId
+		/// - `target`: the LinkTarget containing the asset to check ownership
+		pub fn is_target_owner(
+			who: T::AccountId,
+			target: LinkTarget<T::AccountId>,
+		) -> DispatchResult {
+			match target {
+				LinkTarget::Fragment { definition_hash, edition_id, copy_id } => {
+					let owner = pallet_fragments::Pallet::<T>::get_instance_owner_account_id(
```

Comment:
I would not mix rpc calls with transactions.. can't you just read what you need from storage?

Reply:
Rolled this back. Now pallet-aliases checks Fragment ownership directly from storage.